src/tree_schema.c

```diff
@@ -4454,11 +4454,10 @@ lys_make_implemented_r(struct lys_module *module, struct unres_schema *unres)
 
     for (i = 0; i < module->augment_size; i++) {
         /* apply augment */
-        if (apply_aug(&module->augment[i], unres)) {
+        if ((module->augment[i].flags & LYS_NOTAPPLIED && module->augment[i].target) && apply_aug(&module->augment[i], unres)) {
```

Comment:
Adding LYS_NOTAPPLIED check is the correct fix but why check the augment target? It must be resolved at this point and if it is not, we want to fail the assert in apply_aug(), so please remove it.

Reply:
@michalvasko, I added the target check too because of another edge case we are hitting on certain models; however, we are still trying to truly identify and understand what is going wrong.  I'll remove the target check for now and open a new issue / pull request once we know more about the particular case.

Reply:
Thank you, that would be best for now. Also, if it can safely be reproduced, I will gladly take a look at your use-case myself.